contrib/gitlab/src/mill/contrib/gitlab/GitlabAuthHeaders.scala

```diff
@@ -0,0 +1,11 @@
+package mill.contrib.gitlab
+
+case class GitlabAuthHeaders(headers: Seq[(String, String)])
+
+object GitlabAuthHeaders {
+  def apply(header: String, value: String): GitlabAuthHeaders = GitlabAuthHeaders(Seq(header -> value))
+
+  def personalHeader(token: String): GitlabAuthHeaders = GitlabAuthHeaders("Private-Token", token)
+  def deployHeader(token: String): GitlabAuthHeaders   = GitlabAuthHeaders("Deploy-Token", token)
+  def jobHeader(token: String): GitlabAuthHeaders      = GitlabAuthHeaders("Job-Token", token)
```

Comment:
I would recommend, to stay with the name close the the intended result. E.g. 

```scala
def privateToken(token: String): GitlibAuthHeaders = GitlabAuthHeadsers("Private-Token", token)
```

or 

```scala
def `Private-Token`(token: String): GitlibAuthHeaders = GitlabAuthHeadsers("Private-Token", token)
```

Reply:
Gitlab documentation refers these as personal tokens (https://docs.gitlab.com/ee/user/packages/maven_repository/#authenticate-with-a-personal-access-token-in-maven). I'm a bit dubious if this should be personalToken or privateToken. I do agree, that *Token is a better name.

Reply:
Come to think of it. This is quite "internal" (though not any way hidden). I'll change to privateToken but keep the name `Personal` in `GitlabTokenLookup`.